### ISSUE TITLE
Add message counter to commit log operation 

### ISSUE THREAD

**Issue by bwsw** [bwsw/tstreams-transaction-server#94]
Add counter for every operation, so change (txndata, ts) to (txndata, ts, counter). 
We need a counter to use it during new master election. 

How we are going to use it.
1. if we become a master, we close current CL without MD5 and open new CL.
2. we check if there more or equal of all available peers than floor(N/2), e.g. 1 for 3 (except me, 2 with me).
3. we get all commit logs from that peers and try to build continous sequence 1,2... N

If we achieved it, then it's ok and we can to go. If no, then we have to hang and periodically try to achieve it, because may be new nodes will up.

Why it will work correctly? 

**Because the client must get proper response from server, otherwise it will resend op**, if the current master writes firstly to itself and lastly to clients and next responds. Then when the crash occur before it respond to client, client will resend the op again (to other, new master), so that operation will go to a new commit log again.

So, basically if client release the op, if it's on certain server it's valid and must be used if some node has it, even if the op hasn't persisted to any node, client resend it again to new master. So we must use as much data as we have.

**Comment by bwsw** [comment]
Probably we need to check that preferred master was alive fully during current CL.

**Comment by bwsw** [comment]
Master behaviour:
- master writes to slaves (MWS)
- master writes to itself (MWM)
- master responds to client (MRC)

|MWS | MWM | MRC | Outcome|
-------------------------------------------

**Comment by bwsw** [comment]
The diagram above shows that we never loose the information. Either client will resend it and a new master will get it, or the data were written everywhere.

**Comment by bwsw** [comment]
Record Counter must start from 0 for every CL.
